shared/src/ledger/eth_bridge/bridge_pool_vp.rs

```diff
@@ -131,47 +190,89 @@ where
         let pending: PendingTransfer =
             (&self.ctx).read_post_value(&pending_key)?.ok_or(eyre!(
                 "Rejecting transaction as the transfer wasn't added to the \
-                 pending transfers"
+                 pool of pending transfers"
             ))?;
         if pending != transfer {
             tracing::debug!(
-                "An incorrect transfer was added to the pool: {:?}.\n \
-                 Expected: {:?}",
+                "An incorrect transfer was added to the Ethereum bridge pool: \
+                 {:?}.\n Expected: {:?}",
                 transfer,
                 pending
             );
             return Ok(false);
         }
 
-        // check that gas fees were put into escrow
-
-        // check that the correct amount was deducted from the fee payer
-        if let Some(SignedAmount::Negative(amount)) =
-            self.account_balance_delta(&transfer.gas_fee.payer)
-        {
-            if amount != transfer.gas_fee.amount {
+        // if we are going to mint wNam on Ethereum, the appropriate
+        // amount of Nam must be escrowed in the Ethereum bridge VP's storage.
+        // TODO: We should look this address up from storage
+        if transfer.transfer.asset == wnam() {
+            // check that correct amount of Nam was put into escrow.
+            return if !self.check_nam_escrowed(
+                &transfer.gas_fee.payer,
+                &BRIDGE_POOL_ADDRESS,
+                transfer.gas_fee.amount + transfer.transfer.amount,
+                transfer.gas_fee.amount,
+            ) || !self.check_nam_escrowed(
+                &transfer.transfer.sender,
+                &Address::Internal(InternalAddress::EthBridge),
+                transfer.gas_fee.amount + transfer.transfer.amount,
+                transfer.transfer.amount,
+            ) {
```

Comment:
If someone adds a `PendingTransfer` to the pool, the gas fee payer should have their account debited by `transfer.gas_fee.amount + transfer.transfer.amount`, of which `transfer.gas_fee.amount` goes to the bridge pool. The transfer sender should also have their account debited by `transfer.gas_fee.amount + transfer.transfer.amount`, and `transfer.transfer.amount` should go to the Eth bridge VP (which is acting as the escrow amount here). Shouldn't we be checking `transfer.gas_fee.amount` is debited from the `transfer.gas_fee.payer`, and `transfer.transfer.amount` is debited from `transfer.transfer.sender`, separately?

Reply:
Ooh, yeah, there is an issue here. Need to check if fee payer and gas payer are the same are not. Good catch.